src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Methodologies/MethodologyService.cs

```diff
@@ -123,10 +130,17 @@ public class MethodologyService : IMethodologyService
                 .OnSuccess(_userService.CheckCanAdoptMethodologyForPublication)
                 .OnSuccess(async publication =>
                 {
-                    var methodologies = await _methodologyRepository.GetUnrelatedToPublication(publication.Id);
-                    var latestVersions = await methodologies.SelectAsync(methodology =>
-                        _methodologyVersionRepository.GetLatestVersion(methodology.Id));
-                    return (await latestVersions.SelectAsync(BuildMethodologyVersionViewModel)).ToList();
+                    var publishedMethodologies = await _methodologyRepository
+                        .GetPublishedMethodologiesUnrelatedToPublication(publication.Id);
+                    var latestPublishedVersions = publishedMethodologies
+                        .ToAsyncEnumerable()
+                        .SelectAwait(async methodology =>
+                            await _methodologyVersionRepository.GetLatestPublishedVersion(methodology.Id))
+                        .Where(methodologyVersion => methodologyVersion != null)
+                        .OfType<MethodologyVersion>();
+                    return await latestPublishedVersions
+                        .SelectAwait(async version => await BuildMethodologyVersionViewModel(version))
+                        .ToListAsync();
```

Comment:
To avoid `.Where(methodologyVersion => methodologyVersion != null).OfType<MethodologyVersion>()` You could introduce a new extension method

```
public static IAsyncEnumerable<T> WhereNotNull<T>(this IAsyncEnumerable<T?> source)
    where T : class
{
    return source.Where(item => item is not null)!;
}
```

It would then allow you to replace all this with something like this

```
var publishedMethodologies = await _methodologyRepository.GetPublishedMethodologiesUnrelatedToPublication(publication.Id);

return await publishedMethodologies
    .ToAsyncEnumerable()
    .SelectAwait(async methodology => await _methodologyVersionRepository.GetLatestPublishedVersion(methodology.Id))
    .WhereNotNull()
    .SelectAwait(async version => await BuildMethodologyVersionViewModel(version))
    .ToListAsync();
```